pygmt/tests/test_grdimage.py

```diff
@@ -119,6 +119,20 @@ def test_grdimage_shading_xarray(grid, shading):
     return fig_ref, fig_test
 
 
+@check_figures_equal()
+def test_grdimage_grid_and_shading_with_xarray(grid, xrgrid):
+    """
+    Test that shading works well when xarray.DataArray is input to both the
+    ``grid`` and ``shading`` arguments.
+    """
+    fig_ref, fig_test = Figure(), Figure()
+    fig_ref.grdimage(
+        grid="@earth_relief_01d_g", region="GL", cmap="geo", shading=xrgrid
+    )
+    fig_test.grdimage(grid=grid, region="GL", cmap="geo", shading=xrgrid)
```

Comment:
Remind me again, the xarray shading only works with GMT 6.2 is it? As with issue #364? Here's the diff of Greenland (GL):

| expected | actual (wrong) | diff |
|--|--|--|
| ![test_grdimage_shading_with_xarray png -expected](https://user-images.githubusercontent.com/23487320/106344776-b1e2f600-6310-11eb-9588-6ee4eecfdb36.png) | ![test_grdimage_shading_with_xarray png](https://user-images.githubusercontent.com/23487320/106344775-b0b1c900-6310-11eb-89f0-4f0962d69f83.png) | ![test_grdimage_shading_with_xarray png -failed-diff](https://user-images.githubusercontent.com/23487320/106344777-b27b8c80-6310-11eb-857d-5fa56c1e6a5f.png) |

Reply:
Tried with GMT master. It doesn't work too.

Reply:
Does this branch work now with GMT 6.2.0rc1?

Reply:
Ok, testing with GMT 6.2.0rc1, the PyGMT output seems better than before (when using GMT 6.1.1) as there isn't an obvious diagonal stripe. But expected (baseline) and actual images still look different. This is with `@check_figures_equal` by the way.

| expected | actual (wrong) | diff |
|--|--|--|
| ![test_grdimage_grid_and_shading_with_xarray png -expected](https://user-images.githubusercontent.com/23487320/117589982-62b72780-b181-11eb-9894-51733ddec843.png) | ![test_grdimage_grid_and_shading_with_xarray png](https://user-images.githubusercontent.com/23487320/117589990-6d71bc80-b181-11eb-95b8-760fe8936131.png) | ![test_grdimage_grid_and_shading_with_xarray png -failed-diff](https://user-images.githubusercontent.com/23487320/117589996-7793bb00-b181-11eb-86e3-b99545845852.png) |

Reply:
Ok, I figured out why the images in https://github.com/GenericMappingTools/pygmt/pull/750#discussion_r628957246 are different. For the **baseline** (left) case using `@earth_relief_01d_g`, the CPT is auto-stretched to fit the **Greenland** elevation data range -3893 to 3195.5. For the actual **test** (middle) case using xarray.DataArray, the CPT is auto-stretched to fit the **global** data range -8592.5 to 5559.

Not sure if this can be counted as a minor upstream bug, but a workaround would be to explicitly set a data range for the CPT, or plot a shaded global grid, or maybe try something else.

Reply:
I think it's an upstream bug. It would be easier if you can prepare a minimal example so that @PaulWessel or @meghanrjones can reproduce and debug it.

Reply:
Ok, reported upstream at https://github.com/GenericMappingTools/gmt/issues/5294.